src/pages/Feed/hooks/useCurrentFriend.tsx

```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from 'react';
+import { FriendType, ActionType } from '../types';
+
+const useCurrentFriend = (friendList: FriendType[]) => {
+  const [currentFriend, setCurrentFriend] = useState<FriendType>(friendList[0]);
+  const [action, setAction] = useState<ActionType>(null);
+
+  const handleClickFriend = (userID: string) => {
+    const targetFriend = friendList.find((friend) => friend.userID === userID);
+    if (targetFriend) {
+      setCurrentFriend(targetFriend);
+    }
+  };
+
+  useEffect(() => {
+    //친구를 삭제한 상태이거나 현재 친구가 설정되지 않은 경우
+    if (action === 'delete' || !currentFriend) {
+      setCurrentFriend(friendList[0]);
+    }
+  }, [currentFriend, friendList, action]);
```

Comment:
currentFriend의 값을 직접 참조하여 조건 단순화
```suggestion
  useEffect(() => {
    //친구를 삭제한 상태이거나 현재 친구가 설정되지 않은 경우
    if (!currentFriend) {
      setCurrentFriend(friendList[0] ?? null); // null 값을 명시적으로 지정
    }
  }, [currentFriend, friendList]);
```